src/SymbolicUtils.jl

```diff
@@ -16,12 +16,10 @@ using SymbolicIndexingInterface
 import Base: +, -, *, /, //, \, ^, ImmutableDict
 using ConstructionBase
 using TermInterface
-import TermInterface: iscall, isexpr, issym, symtype, head, children,
+import TermInterface: iscall, isexpr, head, children,
                       operation, arguments, metadata, maketerm
 
-const istree = iscall
-Base.@deprecate_binding istree iscall
```

Comment:
where did the deprecation go?

Reply:
I can restore it, but it looked a bit sketchy... `istree` was defined in TermInterface. It was then renamed to `isexpr`. Why export a function that was deprecated in another package, making it default to another function (and not the one it corresponds to)? Doesn't seem very clean. We could add a deprecation warning in TermInterface if we really want it.

Reply:
The deprecation is required or this is a breaking update.

Reply:
I think this should be another major. Because I also removed the "compatmaker" rewriter hack below

Reply:
Which was marked `XXX remove in future major`

Reply:
> > The deprecation hasn't been around long enough to remove. It should be there at least a year or two.
> 
> Makes sense, but that deprecation seems it shouldn't have been there in the first place... I'm happy to reintroduce it but with the correct function.

But I'm not sure whether changing a deprecation is breaking or not? I would remove, but happy to reintroduce if you think this should really be around

Reply:
Added deprecation